docs: fix php docs parser

RiedleroD opened the pull request:
everything should be working, except default values for optional parameters for some reason

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/necodeIT/lb_planner/80)
<!-- Reviewable:end -->

RiedleroD committed:
try to parse return struct defs

RiedleroD committed:
minor refactors & fixes

RiedleroD committed:
removed embedded script, modify file instead of overwrite

RiedleroD committed:
removed unused import

RiedleroD committed:
a tiny bit of OO

RiedleroD committed:
OO for returns data

RiedleroD committed:
some OO for Parameter Info

RiedleroD committed:
Merge branch 'LP-55-Backend-Refactor' into parse_return_struct_functions

RiedleroD committed:
big steps in getting it to work

RiedleroD committed:
replaced functions' fullname with shortname

RiedleroD committed:
various minor fixes

RiedleroD committed:
ensured return body to be null, not unparseable

RiedleroD committed:
added parsing for enums

RiedleroD committed:
implemented enum parsing for params

RiedleroD committed:
ensured params are either nonempty or null

RiedleroD committed:
mildly better warn system

RiedleroD committed:
fix: put a newline between a warning and its context

RiedleroD committed:
feat: parse optional external_value params in returns as well

also merged the params regex and returns regex into one

RiedleroD committed:
fix: invalid structure definition in notifications helper

RiedleroD committed:
feat: capabilities are now in an array instead of a string

mcquenji reviewed (changes requested):
Moodle CI is failing, rest seems ok

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/necodeIT/lb_planner/80#-:-O2FiZTZC8FkAIB03BW0:b-o2t42q)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @RiedleroD)



<!-- Sent from Reviewable.io -->

mcquenji committed:
feat: check if default is either a valid literal or known special variable

Check default values for known special vars and replace them with a more readable representation.
